controllers/machinedeletion_controller.go

```diff
@@ -63,19 +73,54 @@ func (r *MachineDeletionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if remediation = r.getRemediation(ctx, req); remediation == nil {
 		return ctrl.Result{}, nil
 	}
-	//not a machine based remediation
-	machineOwnerRef := getMachineOwnerRef(remediation)
-	if machineOwnerRef == nil {
-		return ctrl.Result{}, nil
+
+	var machine *unstructured.Unstructured
+	//Health check was done by NHC
+	if node, err := r.getNodeFromMdr(remediation); err == nil {
+		if machine, err = r.buildMachineFromNode(node); err != nil {
+			return ctrl.Result{}, err
+		}
+		if isMachineBelongToMasterNode(machine) {
```

Comment:
Maybe we should do what MHC does here?
https://github.com/openshift/machine-api-operator/blob/edaf826469d651580ddab305962b6a7b5bd6d49a/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go#L811

Reply:
Your suggestion definitely looks more elegant to me.
From how MAO are using it, I believe it should be logically equal to our case but I'm not completely sure.
@beekhof can you take a look please ?